### PR TITLE
flamenco: initial rent collection changes

### PR DESCRIPTION
**Pull request by mjain-jump**

### CODE REVIEW

File: src/flamenco/runtime/fd_runtime.c [hunk 1]
```diff
@@ -3035,46 +3035,40 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
...
+  }
+  else {
+    slots_elapsed = ( epoch - info->rent_epoch) * schedule->slots_per_epoch;
   }
   /* Consensus-critical use of doubles :( */
 
   double years_elapsed;
-  if (FD_LIKELY(slots_per_year != 0.0))
-  {
+  if( FD_LIKELY(slots_per_year != 0.0 ) ) {
```
Comment (kbhargava-jump) [inline src/flamenco/runtime/fd_runtime.c:3067]: `slots_per_year!=0.0`

File: src/flamenco/runtime/fd_runtime.c [hunk 2]
```diff
@@ -3035,46 +3035,40 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
...
+     https://github.com/anza-xyz/agave/blob/v2.0.9/sdk/src/rent_collector.rs#L90 */
+  ulong min_balance = fd_rent_exempt_minimum_balance2( rent, acc->dlen );
+  if( info->lamports >= min_balance ) {
     return FD_RENT_EXEMPT;
   }
 
-  /* Count the number of slots that have passed since last collection */
-
+  /* Count the number of slots that have passed since last collection 
+     https://github.com/anza-xyz/agave/blob/v2.0.9/sdk/src/rent_collector.rs#L93-L98 */
```
Comment (kbhargava-jump) [inline src/flamenco/runtime/fd_runtime.c:3052]: lets use `//` for permalinks instead of `/* */`

File: src/flamenco/runtime/fd_runtime.c [hunk 3]
```diff
@@ -3035,46 +3035,40 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
 #define FD_RENT_EXEMPT (-1L)
 
 static long
-fd_rent_due(fd_account_meta_t *acc,
-            ulong epoch,
-            fd_rent_t const *rent,
-            fd_epoch_schedule_t const *schedule,
-            double slots_per_year)
-{
+fd_rent_due(fd_account_meta_t *         acc,
```
Comment (kbhargava-jump) [inline src/flamenco/runtime/fd_runtime.c:3036]: space after parenthesis and realign

File: src/flamenco/runtime/fd_runtime.c [hunk 4]
```diff
@@ -3035,46 +3035,40 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
...
 
-  /* Nothing due if account is rent-exempt */
-
-  ulong min_balance = fd_rent_exempt_minimum_balance2(rent, acc->dlen);
-  if (info->lamports >= min_balance)
-  {
+  /* Nothing due if account is rent-exempt
+     https://github.com/anza-xyz/agave/blob/v2.0.9/sdk/src/rent_collector.rs#L90 */
+  ulong min_balance = fd_rent_exempt_minimum_balance2( rent, acc->dlen );
+  if( info->lamports >= min_balance ) {
```
Comment (kbhargava-jump) [inline src/flamenco/runtime/fd_runtime.c:3047]: `info->lamports>=min_balance`

**Review by kbhargava-jump — APPROVED** [review]
some nits

File: src/flamenco/runtime/fd_runtime.c (hunk 2 above)
Comment (ibhatt-jumptrading) [reply src/flamenco/runtime/fd_runtime.c:3052]: disagree

File: src/flamenco/runtime/fd_runtime.c [hunk 5]
```diff
@@ -3035,46 +3033,40 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
...
+  }
+  else {
+    slots_elapsed = (epoch - info->rent_epoch) * schedule->slots_per_epoch;
   }
   /* Consensus-critical use of doubles :( */
 
   double years_elapsed;
-  if (FD_LIKELY(slots_per_year != 0.0))
-  {
+  if( FD_LIKELY(slots_per_year!=0.0 ) ) {
```
Comment (kbhargava-jump) [inline src/flamenco/runtime/fd_runtime.c:3067]: spacing

**Review by kbhargava-jump — APPROVED** [review]

**Pull request merged**